docs/spec_draft.md

```diff
@@ -3429,6 +3430,118 @@ More formally, `results[:][j0, ..., jR-1] = reduce(input_slices)` where:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reduce_scatter
+
+### Semantics
+
+<img align="center" src="spec_draft/reduce_scatter.svg" />
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the values of the `operand` tensor from each process,
+splits the reduction result along `scatter_dimension` into parts, and scatters
+the split parts between the processes to produce the `result`.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+  * `channel_id <= 0` and `use_global_device_ids = false`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0` and `use_global_device_ids = false`,
+    `cross_replica_and_partition(replica_groups)`.
+  * `channel_id > 0` and `use_global_device_ids = true`,
+    `flattened_ids(replica_groups)`.
+
+Within each `process_group`:
+  * ```
+    reduced_value@process = reduce-all(operand, replica_groups, channel_handle, use_global_device_ids, computation) for each process in process_group
+    ```
+  * ```
+    split_count = dim(process_groups, 1)
+    split_parts@sender = [
+        slice(
+          operand=operand@sender,
+          start_indices=[s0, s1, ..., sR-1],
+            # where
+            #  - sj = 0 if j != split_dimension
+            #  - sj = i * dim(operand, j) / split_count, if j == split_dimension
+            #  - R = rank(operand)
+          limit_indices=[l0, l1, ..., lR-1],
+            # where
+            #   - lj = dim(operand, j) if j != split_dimension
+            #   - lj = (i + 1) * dim(operand, j) / split_count, if j == split_dimension
+          strides=[1, ..., 1]
+        ) for i in range(split_count)
+    ]
+    ``` for all `sender` in `process_group`.
+  * `result@receiver = split_parts@sender[receiver_index]` for any sender in process_group,
+      where `receiver_index = index_of(receiver, process_group)`.
```

Comment:
Proposed rewording:

```
Afterwards, within each `process_group`:

  * `reduced_value = all_reduce(operand, replica_groups, channel_id, use_global_device_ids, computation)`.
  * `parts@sender = split(operand@sender, dim(process_groups, 1), split_dimension)`.
  * `result@receiver = parts@sender[receiver_index]` for any sender in process_group,
      where `receiver_index = index_of(receiver, process_group)`.
```

Feedback:
  1. Use "afterwards, within each process_group` to be consistent with other collective specs.
  2. Add newline before the bullet list as per #729.
  3. Don't need to say `reduced_value@process` because `all_reduce` is a collective op, so it produces a result on all processes at once.
  4. Let's promote `split` to a meta-function, given that we also need it in `all_to_all`, as well as `convolution`, and it's a hassle to reimplement it every time. This simplifies things considerably.